src/app/shared/services/user-session.service.ts

```diff
@@ -29,18 +30,18 @@ export class UserSessionService implements OnDestroy {
 
   constructor(
     private authService: AuthService,
-    private http: CurrentUserHttpService,
+    private currentUserService: CurrentUserHttpService,
   ) {
-    this.subscription = this.authService.status$.pipe(
-      switchMap(auth => {
-        if (!auth.authenticated) return of<UserProfile|undefined>(undefined);
-        return this.http.getProfileInfo().pipe(
-          catchError(_e => EMPTY)
-        );
-      }),
-      distinctUntilChanged((p1, p2) => p1 === p2 || (!!p1 && !!p2 && p1.groupId === p2.groupId))
+    this.subscription = merge(
+      this.authService.status$.pipe(filter(auth => auth.authenticated)),
+      this.refresh$,
+    ).pipe(
+      switchMap(() => this.currentUserService.getProfileInfo().pipe(
+        catchError(_e => EMPTY)
```

Comment:
I know that was there already but it is strange.  What the best thing to do when it fails?

Reply:
Retry a few times, and if still not working warn through the feedback service. What would you say ?

Reply:
The feedback service UI is not initialized when the user session service constructor is executed, so it won't work as is.
I could still retry a few times, it's not very convincing to me right now.

Reply:
How would it help to warn the user? The problem is that afterwards, we would have, for the platform, the user authenticated but no user profile, an inconsistency which we would have to tackle at many places (and to do what?).
Probably we would have to fallback to the same error as the autherror. (it might just be because the user lost internet connection)

Reply:
Fixed